### ISSUE TITLE
Duplicate rank field number - after creating new entity

### ISSUE THREAD

**Issue by Avinoam200apps** [plantagoIT/strapi-drag-drop-content-type-plugin#46]
Hi,

The bug is quite simple. In my rank field, which currently has values [1, 2, 3], the ordering works perfectly with drag and drop. However, when I add a new entity, I can fill the new rank field with a number that already exists!

For example, if I enter 1 for the rank field, I will end up with two instances of 1 in the position order. Alternatively, if I put 2 in the first position, those two ones will change to become 2! Consequently, it is possible that certain positions will never exist again. As you can see in the picture, position 1 is no longer present.

Thank you for your assistance!

![image](https://github.com/plantagoIT/strapi-drag-drop-content-type-plugin/assets/137150981/81f59f79-ddec-4aec-8d89-5a11468e0ca2)

**Comment by derschiw** [comment]
This is true, the ordering is not complete in that sense. Can you tell me a scenario where this gets an issue? 

I agree that it would be nicer doing this properly. But since it's time consuming I'm not sure if I will fix it for "cosmetic" reasons only, sorry! :thinking:

**Comment by zadinvit** [comment]
It's not just cosmetic problem, because if you sorting it by this rankID and you using pagination and item with same id is last on one page and second is on start of another page, you will get last item twice, because of sort algorithm (is's maybe bug in strapi). Sometimes this plugin reoder thinks automaticly and sometimes not.

**Comment by Mirasaki** [comment]
@zadinvit Do you have a specific reason for not making the `rank` field unique?

**Comment by Mirasaki** [comment]
> @Mirasaki I want it to be unique, but as I report the problem, it sometimes generates some same IDs. It's a good idea to mark it unique, but I don't know if plugin will work with it. I'll test it.

The plugin seems to respect the unique attribute for the `rank` field just fine, we've been running and testing it for a couple of days and haven't encountered any issues with the `unique` attribute applied.

**Comment by Mirasaki** [comment]
Another workaround, when not using `unique`, is to provide a secondary `sort` value when making Strapi API calls: `sort: ['rank:asc','slug:asc'],`

**Comment by zadinvit** [comment]
We solve it this way. We add another parameter to sort.

**Comment by Mirasaki** [comment]
Glad to hear! :)
